pkgs/http_profile/lib/src/http_profile_response_data.dart

```diff
@@ -176,11 +184,12 @@ final class HttpProfileResponseData {
   ///
   /// [endTime] is the time when the response was fully received. It defaults
   /// to the current time.
-  void close([DateTime? endTime]) {
+  Future<void> close([DateTime? endTime]) async {
     _checkAndUpdate();
     _isClosed = true;
-    unawaited(bodySink.close());
-    _data['endTime'] = (endTime ?? DateTime.now()).microsecondsSinceEpoch;
+    await bodySink.close();
+    _responseData['endTime'] =
```

Comment:
So this was incorrect before?

Reply:
It worked before because we were awaiting `_body.stream.expand(...).toList(...)` on the SDK side, so it was impossible for this unawaited `bodySink.close()` to cause the `'requestBody'` or `'responseBody'` entries to get populated incorrectly here (https://github.com/dart-lang/sdk/blob/9efa95674966c1d8b498c1846372f36db689f8c7/sdk/lib/io/network_profiling.dart#L34-L44).

With the changes in this PR, we need to await `bodySink.close()` to guarantee that all the bytes that have been added to the sink have also been added to `_data['requestBodyBytes']` before the SDK tries to read `_data['requestBodyBytes']`.

https://github.com/dart-lang/http/blob/bb1f7a83e4770f6ebc8673a54507833f40eeb433/pkgs/http_profile/lib/src/http_client_request_profile.dart#L81-L83